s3fs/core.py

```diff
@@ -1339,7 +1356,7 @@ def _fetch(self, start, end):
         if start < self.start:
             if not self.fill_cache and end + self.blocksize < self.start:
                 self.start, self.end = None, None
-                return self._fetch(start, end)
```

Comment:
No need to change these - you made sure that `_fetch` does the same thing as before when not mmapping.